modules/tensor_mechanics/include/materials/CreepStressUpdateBase.h

```diff
@@ -0,0 +1,68 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#ifndef CREEPSTRESSUPDATEBASE_H
+#define CREEPSTRESSUPDATEBASE_H
+
+#include "RadialReturnStressUpdate.h"
+
+class CreepStressUpdateBase;
+
+template <>
+InputParameters validParams<CreepStressUpdateBase>();
+
+/**
+ * This class provides baseline functionallity for creep models based on the Discrete material in a
+ * radial return isotropic creep calculations.
+ */
+class CreepStressUpdateBase : public RadialReturnStressUpdate
+{
+public:
+  CreepStressUpdateBase(const InputParameters & parameters);
+
+protected:
+  virtual void initQpStatefulProperties() override;
+  virtual void propagateQpStatefulProperties() override;
+  virtual void computeStressFinalize(const RankTwoTensor & plasticStrainIncrement) override;
+
+  /**
+   * This method returns the derivative of the creep strain with respect to the von mises stress. It
+   * assumes the stress delta (von mises stress used to determine the creep rate) is calculated as:
+   * effective_trial_stress - _three_shear_modulus * scalar
+   */
+  virtual Real computeStressDerivative(const Real effective_trial_stress,
+                                       const Real scalar) override;
+
+  /*
+   * Method that determines the tangent calculation method. For creep only models, the tangent
+   * calculation method is always PARTIAL
+   */
+  virtual TangentCalculationMethod getTangentCalculationMethod() override
+  {
+    return TangentCalculationMethod::PARTIAL;
```

Comment:
@bwspenc This is the other piece that is needed to "turn on" the tangent calculation. All creep models will be `PARTIAL` by definition.